mkdocs/gh_deploy.py

```diff
@@ -5,19 +5,25 @@
 log = logging.getLogger(__name__)
 
 
-def gh_deploy(config):
+def gh_deploy(config, commit_message=None):
 
     if not os.path.exists('.git'):
         log.info('Cannot deploy - this directory does not appear to be a git '
                  'repository')
         return
 
-    log.info("Copying '%s' to `gh-pages` branch and pushing to GitHub.",
-             config['site_dir'])
+    command = ['ghp-import', '-p', config['site_dir']]
+
+    remote_branch = config.get('remote_branch', 'gh-pages')
+    command.extend(['-b', config['remote_branch']])
```

Comment:
@waylan nice catch! I've updated the [defaults](https://github.com/mkdocs/mkdocs/blob/master/mkdocs/config/defaults.py) in config. About the log, I would prefer the log informs the user what branch is being used, just for transparency. Thus the use of the branch name in the log.

@d0ugal since the `remote_branch` is less likely to change, unlike the commit message, I think config setting would suffice. However, I could make another PR to add it to CLI arguments, and maybe add tests for the `gh_deploy` module.